### ISSUE TITLE
pysparksql: Unable to lex characters '${db}'

### ISSUE THREAD

**Issue by asears** [sqlfluff/sqlfluff#3225]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running sqlfluff lint . against a pyspark sql file generated from a Databricks notebook, receive an LXR error and PRS error with widget parameters.

Is there a way to ignore errors or update lexer?  It is valid Databricks pyspark code.

### Expected Behaviour

Code is formatted without error.  This could be an optional warning.

### Observed Behaviour

```plaintext
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'CREATE
                       | WIDGET TEXT partition_day DEFAULT...'
L:  20 | P:  20 |  PRS | Line 20, Position 20: Found unparsable section: ' =
                       | ${partition_day}'
L:  20 | P:  23 |  LXR | Unable to lex characters: '${partitio...'
```

### How to reproduce

Running on Windows.

```shell
sqlfluff lint .
```

sample:

```python
-- COMMAND ----------


CREATE WIDGET TEXT db DEFAULT "";
```

```python
# MAGIC %sql
# MAGIC
# MAGIC select
# MAGIC   'myid' as ID,
# MAGIC   *
# MAGIC from ${db}.mytable limit 1
```

### Dialect

sparksql

### Version

sqlfluff, version 0.13.0
Python 3.9.9

### Configuration

[tool.sqlfluff.core]
templater = "jinja"
sql_file_exts = ".sql,.sql.j2,.dml,.ddl"
dialect = "sparksql"
exclude_rules = "L001"

[tool.sqlfluff.indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = false

[tool.sqlfluff.templater]
unwrap_wrapped_queries = true

[tool.sqlfluff.templater.jinja]
apply_dbt_builtins = true

[tool.sqlfluff.rules]
tab_space_size = 4
max_line_length = 160
indent_unit = "space"
comma_style = "trailing"
allow_scalar = true
single_table_references = "consistent"
unquoted_identifiers_policy = "all"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by asears** [comment]
Thanks for the quick reply.   Seems like this approach "worked" with all the caveats and retesting required.

```shell
sqlfluff fix mycreate.sql --ignore lexing,parsing --FIX-EVEN-UNPARSABLE --force
```

Probably a simple regex addition to the file you mentioned, though I'm not too familiar with the codebase here yet.

Hopefully some additional HiveQL / Databricks Spark SQL syntax comes in the future!
